### ISSUE TITLE
Empty .to([]) sends to everybody instead of nobody

### ISSUE THREAD

**Issue by jyustman** [socketio/socket.io#4645]
**Describe the bug**
`io.to([]).emit("hello")` sends to everybody instead of nobody

**To Reproduce**

Socket.IO server version: `4.6.1`

*Server*

```js
import { Server } from "socket.io";

const io = new Server(3000, {});

io.on("connection", () => {
  io.to([]).emit("hello");
});
```

Socket.IO client version: `4.6.1`

*Client*

```js
import { io } from "socket.io-client";

const socket = io("ws://localhost:3000/", {});

socket.on("hello", () => {
  console.log("hello received");
});
```

**Expected behavior**
I expect no message to be sent when an empty array is passed to `.to()`.

**Platform:**
 - OS: macOS 13.2

**Additional context**
It's easy to work around, but the behavior was unexpected.

**Comment by myYang2016** [comment]
I have this problem too, and I have to check every time to see if it's an empty array

**Comment by darrachequesne** [comment]
@jyustman I could indeed reproduce the behavior, thanks. That is indeed surprising, I guess we can fix this in the next major release.

**Comment by inimaz** [comment]
I can take a look at this one if that's fine?

**Comment by Frank3K** [comment]
I observed the same. Could it be due to the following condition in [socket.io-adapter](https://github.com/socketio/socket.io-adapter/)?

https://github.com/socketio/socket.io-adapter/blob/28f60b880ad04ff4b48ade0cd529f1970643604f/lib/index.ts#L335-L355

**Comment by drmgc** [comment]
@Frank3K it is a in-memory adapter, and indeed it should be smth like `if (rooms.size !== undefined)`. However, I have encountered same issue with a redis adapter.

So, it must be fixed in multiple places

**Comment by darrachequesne** [comment]
@drmgc the issue is that there is no distinction between no room and an empty room:

Source: https://github.com/socketio/socket.io/blob/72d61dab822a2f932b828bbfdc500a566c7cc87f/packages/socket.io/lib/broadcast-operator.ts#L20-L27

**Comment by drmgc** [comment]
@darrachequesne yep, I found out already that `no rooms` is treated as `no specific rooms`.

I thought it would be a simple one-line fix, but properly introducing a "broadcast to [] rooms" requires changes across quite a few places of this codebase, which I'm not familiar with tbh

**Comment by PrasanthGondi** [comment]
Looks like this hasn't been fixed yet. When can we expect a fix for this?
